packages/bpk-component-graphic-promotion/src/BpkGraphicPromo-test.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2022 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import { cssModules } from 'bpk-react-utils';
+
+import BpkGraphicPromo, { TEXT_ALIGN } from './BpkGraphicPromo';
+import type { Props } from './BpkGraphicPromo';
+import STYLES from './BpkGraphicPromo.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+const props: Props = {
+  buttonText: 'Learn more',
+  headline: 'Ride your wave',
+  invertVertically: false,
+  onClick: jest.fn(),
+  sponsor: {
+    label: 'Sponsored',
+    logo: 'path/to/logo.png',
+    altText: 'Airline Name',
+  },
+  subheading:
+    'Portugal and 6 more countries have just been added to the UK travel green list',
+  tagline: 'Tagline',
+  textAlign: TEXT_ALIGN.start,
+};
+
+describe('BpkGraphicPromo', () => {
+  beforeEach(() => {
+    (props.onClick as jest.MockedFunction<Props['onClick']>).mockReset();
```

Comment:
Start TS for mocked functions, otherwise TS will complain that `.mockReset` does not exist on `onClick`.

Reply:
For conciseness, you can use the following, since the exact signature of the mocked function is not needed:

```suggestion
    (props.onClick as jest.Mock).mockReset();
```

Reply:
Knowing that sometimes we need it I like the more verbose form that allows others to easily copy it and quickly infer how to replace when they need to :)